http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/HandlerRemovingChannelPool.java

```diff
@@ -83,7 +83,8 @@ private void removePerRequestHandlers(Channel channel) {
                            FlushOnReadHandler.class,
                            ResponseHandler.class,
                            ReadTimeoutHandler.class,
-                           WriteTimeoutHandler.class);
+                           WriteTimeoutHandler.class,
+                           UnusedChannelExceptionHandler.class);
```

Comment:
Do we still need the IN_USE checks in the handler if we are removing it?

Reply:
Hmm, good point. I'll change it to only add it once when the channel is created and not removing it.

Reply:
Reverted this change. Now we only attach `UnusedChannelExceptionHandler` to h1 channels once they are created